src/app/modules/training-event/components/table-list/table-list.component.ts

```diff
@@ -0,0 +1,59 @@
+import { Component, OnInit, ViewChild } from '@angular/core';
+import { MatPaginator, PageEvent } from '@angular/material';
+import { of as observableOf, Subject, merge } from 'rxjs';
+import { switchMap, map, catchError } from 'rxjs/operators';
+import { CampusEventResponse } from 'src/app/interfaces/event';
+import { EventService  } from '../../services/event.service';
+
+@Component({
+  selector: 'app-table-list',
+  templateUrl: './table-list.component.html',
+  styleUrls: ['./table-list.component.css']
+})
+export class TableListComponent implements OnInit {
+  /** The data to be displayed */
+  events: CampusEventResponse[] = [];
+  /** How to sort the columns */
+  displayedColumns: string[] = ['id', 'name', 'location'];
+  /** The total number of eventlist. */
+  eventlistLength = 0;
+  /** Indicate data loading status */
+  isLoadingResults = true;
+
+  pageSize = 5;
```

Comment:
Pagination size should have been defined in `environment.ts`, do not use custom value if not necessary.